### PR TITLE
Hostname middleware

### PR DESCRIPTION
**Pull request by Joannis**
Adds a middleware that sets the hostname adaptively or statically.

This middleware is not added to the default middleware configuration, if agreed upon I think this should be added.

### CODE REVIEW

File: Tests/VaporTests/MiddlewareTests.swift [hunk 1]
```diff
@@ -24,6 +24,19 @@ class MiddlewareTests : XCTestCase {
         _ = try app.make(Responder.self).respond(to: req).blockingAwait()
         XCTAssert(myMiddleware.flag == true)
     }
+    
+    func testHostnameMiddlware() throws {
```
Comment (tanner0101) [inline Tests/VaporTests/MiddlewareTests.swift]: We need to add an entry to the `allTests` array as well to have this tested on Linux properly.

**Comment by tanner0101** [comment]
Given Vapor will be behind Nginx most of the time, this should at least not be enabled by default in production mode. We could dynamically enable for non-release modes, but that doesn't seem great. I think just offering it is fine.

File: Tests/VaporTests/MiddlewareTests.swift (hunk 1 above)
Comment (0xTim) [reply Tests/VaporTests/MiddlewareTests.swift]: https://github.com/brokenhandsio/SteamPress/blob/vapor3/Tests/SteamPressTests/APITests/APITagControllerTests.swift#L24 can help

Comment (tanner0101) [reply Tests/VaporTests/MiddlewareTests.swift]: @0xTim that is really cool. we should add that to Core

Comment (rafiki270) [reply Tests/VaporTests/MiddlewareTests.swift]: yeah, I already stole it for my tests!!! @0xTim ++

**Pull request closed without merging**